packages/pwa-kit-dev/src/configs/webpack/loaders/extensions-loader.js

```diff
@@ -66,20 +63,32 @@ module.exports = function () {
                         `import ${instanceVariable} from '${modulePath}'`
                 )
                 .join('\n            ')}
-            
+
+            const installedExtensions = [
+                ${extensionDetails
+                    .map(
+                        ({packageName, instanceVariable}) =>
+                            `{packageName: '${packageName}', instanceVariable: ${instanceVariable}}`
+                    )
+                    .join(',\n                ')}
+            ]
+
+            const normalizeExtensionsList = (extensions = []) =>
+                extensions.map((extension) => {
+                    return {
+                        name: Array.isArray(extension) ? extension[0] : extension,
+                        config: Array.isArray(extension) ? {enabled: true, ...extension[1]} : {enabled: true}
+                    }
+                })
+
             export const getExtensions = () => {
-                const configuredExtensions = getConfig()?.app?.extensions || []
+                const configuredExtensions = normalizeExtensionsList(getConfig()?.app?.extensions) || []
+                const enabledExtensions = configuredExtensions.filter((extension) => extension.config.enabled)
 
-                return [
-                    ${extensionDetails
-                        .map(({instanceVariable, packageName}) => {
-                            const config = {} // TODO: Parse config config here.
-                            return `configuredExtensions.includes('${packageName}') ? new ${instanceVariable}(${JSON.stringify(
-                                config
-                            )}) : false`
-                        })
-                        .join(',\n                    ')}
-                ].filter(Boolean)
+                return enabledExtensions.map((extension) => {
+                    const found = installedExtensions.find((ext) => ext.packageName === extension.name)
+                    return found ? new found.instanceVariable(extension.config || {}) : false
+                }).filter(Boolean)
```

Comment:
Just to clarify, you are ensuring that all the configured extensions exist in the installed extensions list, correct?

NIT (style): 

Why didn't you follow the filter pattern through out like this:

```
export const getExtensions = () => {
    const configuredExtensions = 
        (normalizeExtensionsList(getConfig()?.app?.extensions) || [])
            // Only return extensions that are enabled.
            .filter((extension) => extension.config.enabled)
            // Ensure that the extension in the array is actually installed.
            .filter((extension) => installedExtensions.some((ext) => ext.packageName === extension.name))
            // Create the instand with the associated configuration
            .map(() => new found.instanceVariable(extension.config || {}))

    return configuredExtensions
}
```

Reply:
We can't do it like that because the last step (`.map(() => new found.instanceVariable(extension.config || {}))`) is a bit tricky. At that last step, we're working with 2 arrays that are not in the same order. 

In other words, given an extension in the `configuredExtensions`, how do we find its corresponding `instanceVariable` in the other array `installedExtensions`?